core/vibes/soul/lib/streamable.tsx

```diff
@@ -53,6 +53,10 @@ function weakRefCache<K, T extends object>() {
 
 const promiseCache = weakRefCache<string, Promise<unknown>>();
 
+function isResolved<T>(value: Streamable<T>): value is T {
+  return !(value instanceof Promise);
+}
```

Comment:
The naming here is a bit confusing IMO given that promises have a "resolved" state. I first expected us to be checking that the promise was resolved as opposed to checking if something was a promise or not. Perhaps we can name this `isPromise` although honestly, this is the sort of function that I deem unnecessary when the alternative, `value instanceof Promise` vs `isPromise(value)` is the same in terms of mental overhead but without the additional abstraction/function that you have to go read to learn how it works

Reply:
I didn't want to spread `value instanceof Promise` through the implementation, considering that at some point we weren't even sure that `value instanceof Promise` works in all cases.

Reply:
> although honestly, this is the sort of function that I deem unnecessary when the alternative, value instanceof Promise vs isPromise(value) is the same in terms of mental overhead but without the additional abstraction/function that you have to go read to learn how it works

`isPromise(value)` tells you exactly what it does. `value instanceof Promise` is an implementation detail that (a) needs to be deconstructed every time its read to assign it the same meaning and (a) might change in the future. The distinction is subtle but non-negligible.

Reply:
It seems that my latter comments on my opinions on `isPromise` distracted from my main point: I think that `isResolved` is a confusing name and something like `isPromise`, which is more explicit, would be a tad better. In particular, when I read `isResolved` I'm assuming it takes in a promise and returns `true` if the promise has already resolved. Instead, what it does is check that something is not a promise. `!isPromise(value)` would be less confusing in that scenario, IMO.

Reply:
I agree, `isResolved` was a misleading name, I renamed the function to `isPromise`, sorry for not making it clear in the first place.

Reply:
No worries! Thanks for fixing 🙏🏻